pkg/apis/ceph.rook.io/v1/network_test.go

```diff
@@ -72,10 +72,45 @@ func TestValidateNetworkSpec(t *testing.T) {
 	assert.NoError(t, err)
 }
 
-func TestNetworkCephIsHostLegacy(t *testing.T) {
-	net := NetworkSpec{HostNetwork: true}
+// test the NetworkSpec.IsHost method with different network providers
+// Also test it in  combination with the legacy
+// "HostNetwork  setting.
+func TestNetworkCephIsHost(t *testing.T) {
+	net := NetworkSpec{HostNetwork: false}
+
+	net.Provider = NetworkProviderHost
+	ret, err := net.IsHost()
+	assert.NoError(t, err)
+	assert.True(t, ret)
+
+	net.Provider = NetworkProviderDefault
+	net.HostNetwork = true
+	ret, err = net.IsHost()
+	assert.NoError(t, err)
+	assert.True(t, ret)
```

Comment:
Good unit test hygiene is to make sure each test sets up its own inputs so that future test modifications don't have unintended effects. It is hard to read and modify unit tests when test inputs and side-effects are propagated from between test cases.

Please fully specify `net = Network{}` before each test to ensure tests are independent from each other.